jscomp/bin/whole_compiler.ml

```diff
@@ -81072,6 +81072,13 @@ type name =
   | Name_top  of Ident.t
   | Name_non_top of Ident.t
 
+let rec process_comments (el:J.expression list) = match el with 
+| [] -> []
+| e::r -> let new_e = 
+            if e.comment = Some "::" || e.comment = Some "[]"
+            then {J.expression_desc = e.expression_desc; J.comment = None} 
+            else e in
+          new_e::(process_comments r)
 
```

Comment:
this logic is not exact. you can not simply strip "::", you have to check that all its nested are "::"  and replaced the outer-most with "Nested ::".
Yes in OCaml 4.02 "::"" can not be defined by user, but in latest OCaml, "::" could be redefined.
Also `= Some "::"` is not very efficient (due to polymorphic comparison)